src/storefront.rs

```diff
@@ -0,0 +1,371 @@
+use std::collections::HashMap;
+use std::error::Error;
+
+use elementtree::Element;
+use serde::Deserialize;
+
+#[derive(Debug, Default, Deserialize)]
+#[serde(default)]
+pub struct AllowList {
+    pub allowed: Option<Vec<String>>,
```

Comment:
Is there useful meaning in an empty-but-set `allowed` ?  ie. `Some(vec![])` ?  If not, then using `self.allowed.is_empty()` as the trigger rather than `None` would (a) make AllowList smaller and (b) remove a corner case which has no meaning.

On the other hand, if empty-but-set has semantic value, then you should ignore this comment.

Reply:
Yeah, an empty allow list should mean nothing is allowed.